sources/packages/backend/apps/api/src/route-controllers/application-change-request/application-change-request.aest.controller.ts

```diff
@@ -1,19 +1,79 @@
-import { Controller } from "@nestjs/common";
+import {
+  Body,
+  Controller,
+  NotFoundException,
+  Param,
+  ParseIntPipe,
+  Patch,
+} from "@nestjs/common";
+import {
+  AllowAuthorizedParty,
+  Groups,
+  Roles,
+  UserToken,
+} from "../../auth/decorators";
 import { AuthorizedParties } from "../../auth/authorized-parties.enum";
-import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
-import { ApiTags } from "@nestjs/swagger";
-import BaseController from "../BaseController";
-import { ClientTypeBaseRoute } from "../../types";
 import { UserGroups } from "../../auth/user-groups.enum";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { ApiProcessError, ClientTypeBaseRoute } from "../../types";
+import { IUserToken } from "../../auth/userToken.interface";
+import {
+  APPLICATION_NOT_FOUND,
+  ApplicationChangeRequestService,
+} from "../../services";
+import { ApplicationChangeRequestAPIInDTO } from "./models/application-change-request.dto";
+import BaseController from "../BaseController";
+import { Role } from "../../auth";
+import { INVALID_APPLICATION_EDIT_STATUS } from "@sims/services/constants";
 
+/**
+ * Controller for application change request operations for the Ministry.
+ */
 @AllowAuthorizedParty(AuthorizedParties.aest)
 @Groups(UserGroups.AESTUser)
 @Controller("application-change-request")
 @ApiTags(`${ClientTypeBaseRoute.AEST}-application-change-request`)
 export class ApplicationChangeRequestAESTController extends BaseController {
-  constructor() {
+  constructor(
+    private readonly applicationChangeRequestService: ApplicationChangeRequestService,
+  ) {
     super();
   }
 
-  // Add methods later as needed
+  /**
+   * Approves or declines an application change request.
+   * @param applicationId application id.
+   * @param payload request payload.
+   */
+  @Roles(Role.StudentApproveDeclineAppeals)
+  @Patch(":applicationId")
+  @ApiNotFoundResponse({
+    description:
+      "Application to assess change not found or " +
+      "application to assess change not in valid status to be updated.",
+  })
+  async assessApplicationChangeRequest(
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+    @Body() payload: ApplicationChangeRequestAPIInDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
+    try {
+      await this.applicationChangeRequestService.updateApplicationChangeRequestStatus(
+        applicationId,
+        payload.applicationEditStatus,
+        payload.note,
+        userToken.userId,
+      );
+    } catch (error) {
+      if (error.name === APPLICATION_NOT_FOUND) {
+        throw new NotFoundException(error);
+      }
+      throw new NotFoundException(
```

Comment:
This exception must have a check like the one at line 68 otherwise it will be thrown for any error, for instance, a DB connection error.
Please add an `if` and also ensure that any other error will be thrown following the same for other API controllers try/catch.

Reply:
As a suggestion, change the code as below.
```ts
if ([APPLICATION_NOT_FOUND, INVALID_APPLICATION_EDIT_STATUS].includes(error.name)) {
  throw new NotFoundException(error);
}
throw error;
```